lib/instana/instrumentation/instrumented_request.rb

```diff
@@ -9,7 +9,7 @@
 
 module Instana
   class InstrumentedRequest < Rack::Request
-    W3_TRACE_PARENT_FORMAT = /00-(?<trace>[0-9a-f]+)-(?<parent>[0-9a-f]+)-(?<flags>[0-9a-f]+)/.freeze
+    W3C_TRACE_PARENT_FORMAT = /[0-9a-f]{2}-(?<trace>[0-9a-f]{32})-(?<parent>[0-9a-f]{16})-(?<flags>[0-9a-f]{2})(?:-.*)?/.freeze
```

Comment:
Note: Appending the non-capturing group `(?:-.*)?` to the end of the regex and also allowing for other versions than `00` (start of the regex) is the central fix in this PR. The rest is mostly cosmetics. I also went with concrete length for trace id, parent id etc. instead of allowing arbitrary lengths. This is more in line with the w3c trace context specification.